golang/x/relay/types/msgs.go

```diff
@@ -51,3 +53,133 @@ func (msg MsgSetLink) GetSignBytes() []byte {
 
 // Route should return the name of the module
 func (msg MsgSetLink) Route() string { return RouterKey }
+
+/***** IngestHeaderChain *****/
+
+// MsgIngestHeaderChain defines a IngestHeaderChain message
+type MsgIngestHeaderChain struct {
+	Signer  string          `json:"signer"`
+	Headers []BitcoinHeader `json:"headers"`
+}
+
+func NewMsgIngestHeaderChain(address string, headers []BitcoinHeader) MsgIngestHeaderChain {
+	return MsgIngestHeaderChain{
+		address,
+		headers,
+	}
+}
+
+func (msg MsgIngestHeaderChain) GetSigners() []sdk.AccAddress {
+	addr, err := sdk.AccAddressFromBech32(msg.Signer)
+	if err != nil {
+		return []sdk.AccAddress{}
+	}
+	return []sdk.AccAddress{addr}
+}
+
+func (msg MsgIngestHeaderChain) Type() string { return "ingest_header_chain" }
+
+func (msg MsgIngestHeaderChain) ValidateBasic() sdk.Error {
+	for i := range msg.Headers {
+		valid, err := msg.Headers[i].Validate()
+		if valid == false || err != nil {
+			return ErrBadHeaderLength(DefaultCodespace)
```

Comment:
I made a `FromBTCSPVError` exactly for this usecase

Reply:
`return FromBTCSPVError(DefaultCodespace, err)`

Reply:
Ha, right, I found this and then forgot to use it...